src/ios-deploy/ios-deploy.m

```diff
@@ -268,13 +268,16 @@ CFStringRef copy_find_path(CFStringRef rootPath, CFStringRef namePattern) {
         CFRelease(path);
         return NULL;
     }
-    
-    if (CFStringFind(namePattern, CFSTR("/"), 0).location == kCFNotFound) {
-        cf_command = CFStringCreateWithFormat(NULL, NULL, CFSTR("find '%@' -name '%@' -maxdepth 1 2>/dev/null | sort | tail -n 1"), rootPath, namePattern);
-    } else {
-        cf_command = CFStringCreateWithFormat(NULL, NULL, CFSTR("find '%@' -path '%@/%@' 2>/dev/null | sort | tail -n 1"), rootPath, rootPath, namePattern);
+
+    CFIndex maxdepth = 1;
+    CFArrayRef findPathSlash = CFStringCreateArrayWithFindResults(NULL, namePattern, CFSTR("/"), CFRangeMake(0, CFStringGetLength(namePattern)), 0);
+    if (findPathSlash != NULL) {
+        maxdepth = CFArrayGetCount(findPathSlash) + 1;
+        CFRelease(findPathSlash);
     }
 
+    cf_command = CFStringCreateWithFormat(NULL, NULL, CFSTR("find '%@' -path '%@/%@' -maxdepth %ld 2>/dev/null | sort | tail -n 1"), rootPath, rootPath, namePattern, maxdepth);
```

Comment:
We are generally extremely conservative with not breaking existing users for these core use-cases. I think we should fall back to a non-maxdepth version if for some reason nothing is found when specifying the rough maxdepth.

Thanks for looking into this! It will speed up a ton of people's workflows.

Reply:
Curious. Why do we need the flag? Can we always try the new logic first, and if not found, we can fallback to the old logic. In this case we improve the speed for a lof of cases by default and still doesn't regress anyone's usecase. The only drawback I can see with this is that we'll always spend time on both code paths before showing error in the error case. Suggestions? 

Even with the wildcard searches, do we expect a wildcard to match across sub directories in any of cases we have?

Reply:
We should test it, but I'm guessing a lot of the speed is that you aren't currently falling back to the old logic.

If always falling back doesn't kill the speed increases you are seeing than we wouldn't need it.

Reply:
I did miss the fact that this function is called multiple times. And falling back only when the new logic cannot find the path would require much larger refactoring.

Reply:
Update the PR with `--faster-path-search` flag.